### PR TITLE
Spawnable hostile constructs, for away missions and adminfuckery

### PR DESCRIPTION
**Pull request by ChangelingRain**
Adds spawnable npc constructs that will attack targets without the cult faction.
NPC artificers will flee from targets they can't kill in two hits and will also attempt to heal nearby constructs if those constructs are damaged.
Cultists gain and lose the cult faction properly.

Constructs have pressure resistance because flying spooky.
Normal harvesters can break walls and have the cult wall and floor summoning spells.

This should cause no issues with normal constructs, as constructs default to AI_OFF and thus not attacking, moving, or otherwise interacting automatically.

### CODE REVIEW

**Comment by Qbopper** [comment]
I mean considering I literally just started working on an away mission that might involve cult assets, +1

**Comment by Anonus** [comment]
I basically did this entirely for away missions so yeah.

File: code/game/gamemodes/cult/cult.dm [hunk 1]
```diff
@@ -200,6 +201,7 @@
 /datum/game_mode/proc/remove_cultist(datum/mind/cult_mind, show_message = 1)
 	if(cult_mind in cult)
 		cult -= cult_mind
+		cult_mind.current.faction -= "cult"
```
Comment (Anonus) [inline code/game/gamemodes/cult/cult.dm]: "Properly."
This does not actually remove the faction from the list of factions, and from looking around, no faction 'removal' actually seems to remove the faction.

**Comment by WJohn** [comment]
Thank you :+1:

File: code/game/gamemodes/cult/cult.dm (hunk 1 above)
Comment (Supermichael777) [inline code/game/gamemodes/cult/cult.dm]: i think he means deconversions make cultists no longer in the faction list, not that the faction gets deleted

Comment (Anonus) [inline code/game/gamemodes/cult/cult.dm]: This does not remove the cult faction from the mob being deconverted, for clarity. I'm confused as to why it doesn't work.

File: code/modules/mob/living/simple_animal/constructs.dm [hunk 2]
```diff
@@ -177,10 +187,52 @@
 						<i>and, most important of all,</i> create new constructs by producing soulstones to capture souls, \
 						and shells to place those soulstones into.</b>"
 
+/mob/living/simple_animal/hostile/construct/builder/Found(atom/A) //what have we found here?
+	if(isliving(A))
```
Comment (duncathan) [inline code/modules/mob/living/simple_animal/constructs.dm]: why can't this check just be the istype you do two lines down?

File: code/modules/mob/living/simple_animal/constructs.dm [hunk 3]
```diff
@@ -177,10 +187,52 @@
...
+
+/mob/living/simple_animal/hostile/construct/builder/CanAttack(atom/the_target)
+	if(see_invisible < the_target.invisibility)//Target's invisible to us, forget it
+		return 0
+	if(Found(the_target) || ..()) //If we Found it or Can_Attack it normally, we Can_Attack it as long as it wasn't invisible
+		return 1 //as a note this shouldn't be added to base hostile mobs because it'll mess up retaliate hostile mobs
+
+/mob/living/simple_animal/hostile/construct/builder/MoveToTarget(var/list/possible_targets)
+	..()
+	if(target && isliving(target))
```
Comment (duncathan) [inline code/modules/mob/living/simple_animal/constructs.dm]: see above. you can also remove the check for `target`, as null will fail the istype.

File: code/modules/mob/living/simple_animal/constructs.dm [hunk 4]
```diff
@@ -177,10 +187,52 @@
...
+		if(istype(L, /mob/living/simple_animal/hostile/construct) && L.health >= L.maxHealth) //is this target an unhurt construct? stop trying to heal it
+			LoseTarget()
+			return 0
+		if(L.health <= melee_damage_lower+melee_damage_upper) //ey bucko you're hurt as fuck let's go hit you
+			retreat_distance = null
+			minimum_distance = 1
+
+/mob/living/simple_animal/hostile/construct/builder/Aggro()
+	..()
+	if(target && istype(target, /mob/living/simple_animal/hostile/construct)) //oh the target is a construct no need to flee
```
Comment (duncathan) [inline code/modules/mob/living/simple_animal/constructs.dm]: see above.

File: code/modules/mob/living/simple_animal/constructs.dm (hunk 2 above)
Comment (Anonus) [inline code/modules/mob/living/simple_animal/constructs.dm]: honking noise

File: code/modules/mob/living/simple_animal/constructs.dm (hunk 4 above)
Comment (Anonus) [inline code/modules/mob/living/simple_animal/constructs.dm]: clown sound

File: code/modules/mob/living/simple_animal/constructs.dm (hunk 3 above)
Comment (Anonus) [inline code/modules/mob/living/simple_animal/constructs.dm]: airhorn blast

File: code/game/gamemodes/cult/cult.dm (hunk 1 above)
Comment (Anonus) [inline code/game/gamemodes/cult/cult.dm]: NOBODY KNOWS WHY THIS FAILS, I'M GOING TO PRETEND IT'S FINE

**Pull request merged**